File: app/src/main/java/com/example/architectureapplication/di/MainComponent.java
package com.example.architectureapplication.di;

import com.example.architectureapplication.MainActivityKt;

import dagger.Subcomponent;

@Subcomponent()
public interface MainComponent {

    @Subcomponent.Builder
    interface Builder{
        MainComponent build();
    }

    void inject(MainActivityKt mainActivity);
}

File: app/src/main/java/com/example/architectureapplication/kt/MainComponent.kt
package com.example.architectureapplication.kt

import com.example.architectureapplication.MainActivity
import dagger.Subcomponent

@Subcomponent
interface MainComponent {
    fun inject(MainActivity: MainActivity)

    @Subcomponent.Builder
    interface Builder{
        fun build(): MainComponent
    }
}
